sentry_sdk/integrations/opentelemetry/sampler.py

```diff
@@ -0,0 +1,105 @@
+from random import random
+
+from opentelemetry import trace
+from opentelemetry.sdk.trace.sampling import Sampler, SamplingResult, Decision
+from opentelemetry.trace import SpanKind
+from opentelemetry.util.types import Attributes
+
+import sentry_sdk
+from sentry_sdk.integrations.opentelemetry.consts import SENTRY_TRACE_STATE_DROPPED
+from sentry_sdk.tracing_utils import has_tracing_enabled
+from sentry_sdk.utils import is_valid_sample_rate, logger
+
+from typing import Optional, Sequence
+
+
+def get_parent_sampled(parent_context, trace_id):
+    # type: (Optional["Context"], int) -> Optional[bool]
+    is_span_context_valid = parent_context is not None and parent_context.is_valid
+
+    # Only inherit sample rate if `traceId` is the same
+    if is_span_context_valid and parent_context.trace_id == trace_id:
+        # this is getSamplingDecision in JS
+        if parent_context.trace_flags.sampled:
+            return True
+
+        dropped = parent_context.trace_state.get(SENTRY_TRACE_STATE_DROPPED) == "true"
+        if dropped:
+            return False
+
+        # TODO-anton: fall back to sampling decision in DSC (for this die DSC needs to be set in the trace_state)
+
+    return None
+
+
+class SentrySampler(Sampler):
+    def should_sample(
+        self,
+        parent_context: Optional["Context"],
+        trace_id: int,
+        name: str,
+        kind: Optional[SpanKind] = None,
+        attributes: Attributes = None,
+        links: Optional[Sequence["Link"]] = None,
+        trace_state: Optional["TraceState"] = None,
+    ) -> "SamplingResult":
+        client = sentry_sdk.get_client()
+
+        # No tracing enabled, thus no sampling
+        if not has_tracing_enabled(client.options):
+            return SamplingResult(Decision.DROP)
+
+        # Check if sampled=True was passed to start_transaction
+        # TODO-anton: Do we want to keep the start_transaction(sampled=True) thing?
+
+        parent_span = trace.get_current_span()
```

Comment:
```suggestion
        parent_span = trace.get_current_span(parent_context)
```

usually `parent_context` is `None` so this falls back to your code 99% of the time, but there are use cases where it might be passed explicitly, so we need to use that

Reply:
done.